### ISSUE TITLE
False Positives on 2 phones and misleading information about when the encounter happened

### ISSUE THREAD

**Issue by tolzhabayev** [corona-warn-app/cwa-app-android#2000]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Technical details
First device:
- Device name: Pixel 5
- Android version: Android 11
- App version: 1.9.1
- ENF Version: 18204713000
- Rooted: NO
- ENF LOG: 
[all-exposure-checks-pixel3g.txt](https://github.com/corona-warn-app/cwa-app-android/files/5738752/all-exposure-checks-pixel3g.txt)

Second device:
- Device name: Pixel 3a
- Android version: Android 11
- App version: 1.9.1
- ENF Version: 18204516000
- Rooted: NO
- ENF Log: 
[all-exposure-checks-pixel5.txt](https://github.com/corona-warn-app/cwa-app-android/files/5738756/all-exposure-checks-pixel5.txt)

## Describe the bug

Since yesterday, 23.12.2020 around 21:00 CEST the app shows on **Pixel 3g** that there was a high risk contact within the last 3 days. 

**Exact wording:**
"Sie haben ein erhöhtes Infektionsrisiko, da Sie zuletzt vor 3 Tagen mindestens einer nachweislich Corona-positiv getesteten Person über einen längeren Zeitraum und mit geringen Abstand begegnet sind". See screenshot for that.

At the same time the app says - "Zuletzt am 15.12.20202" 

This is confusing for the user, which information is now **correct** either 15.12. or the last 3 days?

Now since today, 24.12.2020 around 08:00 CEST the **Pixel 5** shows "Begegnung an 4 Tagen mit erhöhtem Risiko" and "Zuletzt am 17.12.2020". The detail text says:

"Sie haben ein erhöhtes Infektionsrisiko, da Sie zuletzt vor 4 Tagen mindestens einer nachweislich Corona-positiv getesteten Person über einen längeren Zeitraum und mit geringen Abstand begegnet sind". See screenshot for that."

This is confusing for the user, which information is now **correct** either 17.12. or the last 4 days?

**Now the next weird thing for Pixel 3g**
When looking through the ENA logs (attached above) you can see for Pixel 3g that it has 0! matchesCount except for ONLY ONE time on `23. Dezember 2020, 20:27` which shows up `5`. Even after that it shows 0 which is as far as I can read from the other issues here on github is not how it works. How come there were 0 other matchesCount before and **AFTER** this date?

**Same weird thing for Pixel 5**:
When looking through the ENA logs (attached above) you can see for Pixel 5g that it has:
- 1 match `17.12. 00:12 till 21.12. 15:17` 
- 0 matches between `21.12. 19:18 till 23.12. 19:50` 
- 5 matches exactly once on `24.12. 02:01`
- 0 matches following after that for example on `24.12. 06:04`

How is it possible, shouldn't there be 5 matches also in the next checks until they "expire"?

## Additional context
- On 17.12.2020 Pixel 5 was at all times isolated at home, no outside contacts whatsoever, how is it possible?
- On 15.12.2020 Pixel 3g was at all times isolated at home, no outside contacts whatsoever and also has 0 matches in ENF log, how come the app shows risk contact at that day?
- There seems to be a false positive report / server bug which shows 5 matches on both phones?
- If there is no bug and the information is correct - which one? The dates 17.12. and 15.12. or "the last 3/4 days"?
- Unfortunately both phones are not rooted so can not get more info here

## Screenshots
# Pixel 3g:
![WhatsApp Image 2020-12-24 at 08 57 39](https://user-images.githubusercontent.com/580672/103073853-a6d9cf00-45c8-11eb-8a56-ed082a19d0c1.jpeg)
![WhatsApp Image 2020-12-24 at 08 59 47](https://user-images.githubusercontent.com/580672/103073855-a80afc00-45c8-11eb-952e-dece687d39cf.jpeg)

# Pixel 5:
![Screenshot_20201224-090252](https://user-images.githubusercontent.com/580672/103073869-ae00dd00-45c8-11eb-9e89-f22eca2c2e3d.png)

**Comment by MikeMcC399** [comment]
@tolzhabayev 
One of your issues regarding the conflicting information about when an encounter took place was already reported in https://github.com/corona-warn-app/cwa-app-android/issues/1964 which is labelled to be fixed in version 1.10.

**Comment by tolzhabayev** [comment]
@MikeMcC399 thanks so that would mean 15.12. for Pixel 3a and 17.12. for Pixel 5 should be the "official" dates of encounter, right?

Then it still remains unclear how come pixel 3 has 0 matches till 23.12., then 5 matches once on 23.12. and then 0 matches afterwards.

Same for Pixel 5 - no matches on 21.12. - 23.12 and then suddenly the 5 matches at 24.12. again only once so not really clear if they were high or low level risk.

Both phones have been close to each other for the last 2 weeks at least which is very weird for them to have different data.

**Comment by MikeMcC399** [comment]
@tolzhabayev 
> so that would mean 15.12. for Pixel 3a and 17.12. for Pixel 5 should be the "official" dates of encounter, right?

According to the PR https://github.com/corona-warn-app/cwa-app-android/pull/1974 it's the long text starting "Infektionsrisiko So wird Ihr Risiko ermittelt" which contains the incorrect value, so I agree that you should use the absolute dates 15.12. (Pixel 3a) and 17.12. (Pixel 5).

> On 15.12.2020 Pixel 3g (sic) was at all times isolated at home, no outside contacts whatsoever and also has 0 matches in ENF log, how come the app shows risk contact at that day?
>
> On 17.12.2020 Pixel 5 was at all times isolated at home, no outside contacts whatsoever, how is it possible?

I can't see any explanation for a phone declaring later an increased risk for a day that it was isolated.

Do you know when your Pixel 3a updated to CWA 1.9.1? The roll-out made the app available to all Android users by Dec 21, 2020. 

CWA 1.9.1 uses a completely different risk calculation method based on v2 of the [Google Exposure Notifications System](https://developers.google.com/android/exposure-notifications/exposure-notifications-api). There is an [FAQ article](https://www.coronawarn.app/en/faq/#encounter_count_19) which says that the display of low risk encounters may change after installing CWA 1.9.1. It does not discuss increased risk encounters though. There might be some other calculation glitches due to the change of risk algorithm change which aren't well understood at the moment.

**Comment by tolzhabayev** [comment]
> Do you know when your Pixel 3a updated to CWA 1.9.1? The roll-out made the app available to all Android users by Dec 21, 2020.

It was updated via auto-update, elixir 2 says it was last updated on 18.12.2020. Are you sure about 21.12.2020 - the google play store page (at the bottom) says 17.12.2020 (https://play.google.com/store/apps/details?id=de.rki.coronawarnapp)

Just for my sanity - regardless of the Google Exposure Notification System version - shouldn't the Pixel 3a have more `matchesCount` then the only one it has?

I assumed it would act like following:
- On 15.12.2020 or latest 16.12.2020 it should have at least 1 match. But it has 0 on ALL the entries except the one on 23.12.2020.
- On 24.12.2020 - 03:07 and next syncs going forward for a couple of days it should have constantly show 5 matches - but it shows 0.

Or am I missing something?

**Comment by MikeMcC399** [comment]
@tolzhabayev 
>>Do you know when your Pixel 3a updated to CWA 1.9.1? The roll-out made the app available to all Android users by Dec 21, 2020.

> It was updated via auto-update, elixir 2 says it was last updated on 18.12.2020. Are you sure about 21.12.2020 - the google play store page (at the bottom) says 17.12.2020 (https://play.google.com/store/apps/details?id=de.rki.coronawarnapp)

CWA Android 1.9.1 was a staged rollout (which means that different Google Accounts got access to the version at different times) - see https://www.coronawarn.app/en/blog/2020-12-16-corona-warn-app-version-1-9/ . It was announced on Dec 16, uploaded on Dec 17, approved by Google on Dec 18 and most people had got it by Dec 20. Your date of Dec 18 fits perfectly. Also knowing the date of update means that we can assume a certain stability in the exposure checks from that day onwards.

> I assumed it would act like following:
>
> On 15.12.2020 or latest 16.12.2020 it should have at least 1 match. But it has 0 on ALL the entries except the one on 23.12.2020.
On 24.12.2020 - 03:07 and next syncs going forward for a couple of days it should have constantly show 5 matches - but it shows 0.
Or am I missing something?

I'm not suspicious about the 0 matches in your Pixel 3a log for 15. Dez 02:42 to 23. Dez 15:41. Those timestamps correspond to the times that the check was made. They are not the time of the encounter. If the user reported the infection on 23. Dez then CWA will only show matches in the log after the infection was reported.

I do think it is suspicious for the match count to go from 5 at 23. Dez 20:27 to 0 at 24. Dez, given that CWA tells you the encounter happened on 15. Dez. It's really difficult to work this out though because we don't have access to the raw data being used in the calculation.

The latest diagram for the risk calculation is on https://github.com/corona-warn-app/cwa-documentation/blob/master/images/risk_calculation/risk_calculation_enf_v2_overview.pdf . Unfortunately there is no text description to go with it yet. We had some questions about it which were discussed in https://github.com/corona-warn-app/cwa-documentation/issues/485.

I doubt if we can find out more about the issue you have reported (apart from the known issue which is potentially fixed in PR https://github.com/corona-warn-app/cwa-app-android/pull/1974) until one of the technical experts from the Open Source Team is able to contribute.

**Comment by heinezen** [comment]
Since version 1.10 has been released to the public, did the strange behaviour happen to you again @tolzhabayev ?

----
Corona-Warn-App Open Source Team

**Comment by tolzhabayev** [comment]
> Since version 1.10 has been released to the public, did the strange behaviour happen to you again @tolzhabayev ?
> 
> Corona-Warn-App Open Source Team

I have now 1.10.1 and the status is switched to green now with "no contacts" - don't really know if it is a good or a bad thing tbh. I think we can close this bug ticket for now but this behavior might unsettle some people who do not really understand how it works.

**Comment by MikeMcC399** [comment]
@tolzhabayev 
>  I think we can close this bug ticket for now but this behavior might unsettle some people who do not really understand how it works.

I'm sure you are right that the change in results may cause some people some worry. That was the reason for creating the FAQ article:

https://www.coronawarn.app/en/faq/#encounter_count_19

and for a deeper explanation:

https://www.coronawarn.app/en/faq/#encounter_19_calc

The change only happens once after the update to version 1.9 or later, then the calculation method stays stable.

Since you are the author, you can close the issue yourself if you want to, using the button at the bottom.
